### DISCUSSION TITLE
Сonnection pool

### DISCUSSION THREAD

**Discussion by Pingear** [isoos/postgresql-dart#259] [Q&A]
Hi all)
1) Please tell me how to configure the connection pool.
2) In the documentation, all that is said about the pool is “The library supports connection pooling (and masking the connection pool as regular session executor)”. Does the fact that “masking the connection pool” indicate that the connection pool is enabled by default or only after activating this functionality does “masking” occur?

**Comment by isoos** [comment]
On the example page we have the following code to create a simple connection:

```dart
  final conn = await Connection.open(
    Endpoint(
      host: 'localhost',
      database: 'postgres',
      username: 'user',
      password: 'pass',
    ),
    // The postgres server hosted locally doesn't have SSL by default. If you're
    // accessing a postgres server over the Internet, the server should support
    // SSL and you should swap out the mode with `SslMode.verifyFull`.
    settings: ConnectionSettings(sslMode: SslMode.disable),
  );
  
  // ...
  
  final result2 = await conn.execute(
    Sql.named('SELECT * FROM a_table WHERE id=@id'),
    parameters: {'id': 'example row'},
  );
```
The same thing with connection pooling:

```dart
  final pool = await Pool.withEndpoints(
    [Endpoint(
      host: 'localhost',
      database: 'postgres',
      username: 'user',
      password: 'pass',
    )],
    // The postgres server hosted locally doesn't have SSL by default. If you're
    // accessing a postgres server over the Internet, the server should support
    // SSL and you should swap out the mode with `SslMode.verifyFull`.
    settings: PoolSettings(sslMode: SslMode.disable),
  );
  
  // ...
  
  final result2 = await pool.execute(
    Sql.named('SELECT * FROM a_table WHERE id=@id'),
    parameters: {'id': 'example row'},
  );
```
The configuration options are in `PoolSettings` (which is an extended `ConnectionSettings`). I have only changed the `Connection.open` to `Pool.withEndpoints` and referenced the `pool` variable instead of `conn` to execute a query.

In another words, `Pool` and `Connection` can be used almost always interchangeably as long as you are referencing them as `Session` or `SessionExecutor` (they implement both - maybe we should have a common interface for these methods... or `SessionExecutor` also extend `Session`).

Does this give a good-enough starting point? Note: we are not that strong on beginner-friendly documentation, PRs are accepted.
